adjusted BMI for master files

caitlink12 opened the pull request:
added HTWDBMI_der (derived BMI) calculated with HTWDHTM (continuous height) HWTDWTK (continuous weight) for master files 2001-2018 into variable_details
added HWTDCOR_der (corrected BMI) calculated using HWTDHTM (continuous height) and HWTDWTK (continuous weight) for master files 2001-2018 into variable_details
added HTWDBMI_der and HWTDCOR_der along with necessary information in variables.csv

caitlink12 committed:
temp

caitlink12 committed:
derived adjusted BMI

included 2001 into derived adjusted BMI (grouped)

caitlink12 committed:
adjusted derived BMI

created adjusted derived BMI variable with true height and weight values for ICES surveys

caitlink12 committed:
temp

caitlink12 committed:
remove 2001 from HWTGCOR

2001_m has non-grouped height and weight variables to use for HWTDCOR so do not need to include it in HWTGCOR

caitlink12 committed:
included derived BMI variables

added in HWTDBMI_der, HWTDBMI_der_cat4, and HWTDCOR_der for master files to use continuous height and weight values

yulric committed:
fix big diffs in files

yulric reviewed (changes requested):
@caitlink12 I'm not sure what happened but it looks like the changes for the new BMI variables in the variables details sheet are not there (?). Do you mind re-adding them.

caitlink12 committed:
track changes

adjusting to track changes properly

caitlink12 committed:
BMI variables

updated HWTDBMI, HWTDBMI_der, HWTDCOR_der variables which use continuous height and weight measures

caitlink12 committed:
Merge remote-tracking branch 'origin/adj-bmi' into adj-bmi

caitlink12 commented:
@yulric the variables were still in the sheet but the changes must not have been tracked after I had to make additional adjustments to another variables after the initial push. I've re-added them in now so they should appear as a tracked change now.

yulric reviewed (changes requested):
I still can't see the changes in the variable details sheet on Github but that's ok, I looked at the rows within the variable details sheet. The only issue I see is that the database suffixes are inconsistent between the two sheets.

caitlink12 committed:
suffixes

corrected _i suffixes to _m for HWTDBMI, HWTDBMI_der, HWTDCOR_der, HWTDHTM, HWTDWTK

caitlink12 commented:
@yulric I've adjusted the suffixes to _m for HWTDBMI, HWTDBMI_der, HWTDCOR_der, HWTDHTM and HWTDWTK.

rafdoodle committed:
Corrected databaseStart and variableStart for HWT variables

rafdoodle commented on inst/extdata/variable_details.csv:
```diff
@@ -1609,56 +1586,63 @@ HUPDPAD,HUPDPAD_cat5_5,cat,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p,
+HWTDBMI,N/A,cont,"cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2009_m, cchs2010_m, cchs2011_2012_m, cchs2012_m, cchs2013_2014_m, cchs2015_2016_m, cchs2017_2018_m","cchs2005_m::HWTEDBMI, cchs2015_2016_m::HWTDVBMI, cchs2017_2018_m::HWTDVBMI, [HWTDBMI]",ICES specific,cont,NA::a,N/A,not applicable,not applicable,kg/m2,999.96,Not applicable,BMI,BMI / self-report - (D),,,,active,v3.0.0alpha,
+HWTDBMI,N/A,cont,"cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2009_m, cchs2010_m, cchs2011_2012_m, cchs2012_m, cchs2013_2014_m, cchs2015_2016_m, cchs2017_2018_m","cchs2005_m::HWTEDBMI, cchs2015_2016_m::HWTDVBMI, cchs2017_2018_m::HWTDVBMI, [HWTDBMI]",ICES specific,cont,NA::b,N/A,missing,missing,kg/m2,"[999.97,999.99]",don't know (999.97); refusal (999.98); not stated (999.99),BMI,BMI / self- ...
+HWTDBMI,N/A,cont,"cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2009_m, cchs2010_m, cchs2011_2012_m, cchs2012_m, cchs2013_2014_m, cchs2015_2016_m, cchs2017_2018_m","cchs2001_m::HWTADBMI, cchs2003_m::HWTCDBMI, cchs2005_m::HWTEDBMI, cchs2015_2016_m::HWTDVBMI, cchs2017_2018_m::HWTDVBMI, [HWTDBMI]",ICES specific,cont,NA::b,N/A,missing,missing,kg/m2,else,else,BMI,BMI / self ...
+HWTDBMI_der,N/A,cont,"cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2009_m, cchs2010_m, cchs2011_2012_m, cchs2012_m, cchs2013_2014_m, cchs2015_2016_m, cchs2017_2018_m","DerivedVar::[HWTDHTM, HWTDWTK]",,N/A,Func::calculate_bmi_D,N/A,N/A,N/A,kg/m2,N/A,N/A,Derived BMI,Derived Body Mass Index,,,,active,,No typeStart
```
`_D` functions do not exist in `R/bmi.R` -> need to be added

rafdoodle commented on inst/extdata/variable_details.csv:
```diff
@@ -1609,56 +1586,63 @@ HUPDPAD,HUPDPAD_cat5_5,cat,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p,
+HWTDBMI,N/A,cont,"cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2009_m, cchs2010_m, cchs2011_2012_m, cchs2012_m, cchs2013_2014_m, cchs2015_2016_m, cchs2017_2018_m","cchs2001_m::HWTADBMI, cchs2003_m::HWTCDBMI, cchs2005_m::HWTEDBMI, cchs2015_2016_m::HWTDVBMI, cchs2017_2018_m::HWTDVBMI, [HWTDBMI]",ICES specific,cont,NA::b,N/A,missing,missing,kg/m2,else,else,BMI,BMI / self ...
+HWTDBMI_der,N/A,cont,"cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2009_m, cchs2010_m, cchs2011_2012_m, cchs2012_m, cchs2013_2014_m, cchs2015_2016_m, cchs2017_2018_m","DerivedVar::[HWTDHTM, HWTDWTK]",,N/A,Func::calculate_bmi_D,N/A,N/A,N/A,kg/m2,N/A,N/A,Derived BMI,Derived Body Mass Index,,,,active,,No typeStart
+HWTDBMI_der,N/A,cont,"cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2009_m, cchs2010_m, cchs2011_2012_m, cchs2012_m, cchs2013_2014_m, cchs2015_2016_m, cchs2017_2018_m","DerivedVar::[HWTDHTM, HWTDWTK]",,N/A,NA::b,N/A,missing,missing,kg/m2,N/A,N/A,Derived BMI,Derived Body Mass Index,,,,active,,No typeStart
+HWTDBMI_der_cat4,HWTDBMI_der_cat4N/A_Func::categorize_bmi_D,cat,"cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2009_m, cchs2010_m, cchs2011_2012_m, cchs2012_m, cchs2013_2014_m, cchs2015_2016_m, cchs2017_2018_m",DerivedVar::[HWTDBMI_der],,N/A,Func::categorize_bmi_D,N/A,N/A,N/A,kg/m2,N/A,N/A,Categorical BMI,Categorical body mass index,,,,discontinued,,No typeStart
```
- `categorize_` functions removed from `R/bmi.R` -> `cat4` rows in `variable_details.csv` need to be refactored
- `_D` functions do not exist in `R/bmi.R` -> they need to be added

rafdoodle commented on inst/extdata/variable_details.csv:
```diff
@@ -1609,56 +1586,63 @@ HUPDPAD,HUPDPAD_cat5_5,cat,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p,
+HWTDBMI_der_cat4,HWTDBMI_der_cat4N/A_Func::categorize_bmi_D,cat,"cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2009_m, cchs2010_m, cchs2011_2012_m, cchs2012_m, cchs2013_2014_m, cchs2015_2016_m, cchs2017_2018_m",DerivedVar::[HWTDBMI_der],,N/A,4,4,Obese,Obese with BMI greater than 30,kg/m2,4,Obese,Categorical BMI,Categorical body mass index,,,,discontinued,,
+HWTDBMI_der_cat4,HWTDBMI_der_cat4N/A_Func::categorize_bmi_D,cat,"cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2009_m, cchs2010_m, cchs2011_2012_m, cchs2012_m, cchs2013_2014_m, cchs2015_2016_m, cchs2017_2018_m",DerivedVar::[HWTDBMI_der],,N/A,NA::a,4,not applicable,not applicable,kg/m2,NA::a,not applicable,Categorical BMI,Categorical body mass index,,,,discontinued,,
+HWTDBMI_der_cat4,HWTDBMI_der_cat4N/A_Func::categorize_bmi_D,cat,"cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2009_m, cchs2010_m, cchs2011_2012_m, cchs2012_m, cchs2013_2014_m, cchs2015_2016_m, cchs2017_2018_m",DerivedVar::[HWTDBMI_der],,N/A,NA::b,4,missing,missing,kg/m2,NA::b,missing,Categorical BMI,Categorical body mass index,,,,discontinued,,
+HWTDCOR_der,N/A,cont,"cchs2001_m, cchs2003_m, cchs2005_m, cchs2007_2008_m, cchs2009_2010_m, cchs2009_m, cchs2010_m, cchs2011_2012_m, cchs2012_m, cchs2013_2014_m, cchs2015_2016_m, cchs2017_2018_m","DerivedVar::[DHH_SEX, HWTDHTM, HWTDWTK]",,N/A,Func::adjust_bmi_D,N/A,N/A,N/A,kg/m2,N/A,N/A,Derived adjusted BMI,Derived adjusted BMI,,,,active,,No typeStart
```
`_D` functions do not exist in `R/bmi.R` -> they need to be added

rafdoodle commented on inst/extdata/variable_details.csv:
```diff
@@ -1609,56 +1586,63 @@ HUPDPAD,HUPDPAD_cat5_5,cat,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p,
+HWTGBMI,N/A,cont,"cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p","cchs2005_p::HWTEGBMI, cchs2015_2016_p::HWTDGBMI, cchs2017_2018_p::HWTDGBMI, [HWTGBMI]",ICES confirmed,cont,NA::a,N/A,not applicable,not applicable,kg/m2,999.96,Not applicable,BMI,"BMI / self-report - (D,G)",,,,active,v3.0.0alpha,
+HWTGBMI,N/A,cont,"cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p","cchs2005_p::HWTEGBMI, cchs2015_2016_p::HWTDGBMI, cchs2017_2018_p::HWTDGBMI, [HWTGBMI]",ICES confirmed,cont,NA::b,N/A,missing,missing,kg/m2,"[999.97,999.99]",don't know (999.97); refusal (999.98); not stated (999.99),BMI,"BMI / sel ...
+HWTGBMI,N/A,cont,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p","cchs2001_p::HWTAGBMI, cchs2003_p::HWTCGBMI, cchs2005_p::HWTEGBMI, cchs2015_2016_p::HWTDGBMI, cchs2017_2018_p::HWTDGBMI, [HWTGBMI]",ICES confirmed,cont,NA::b,N/A,missing,missing,kg/m2,else,else,BMI,"BMI / se ...
+HWTGBMI_cat4,HWTGBMI_cat4N/A_range,cat,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p",DerivedVar::[HWTGBMI],,cat,copy,4,Categorical BMI,Categorical body mass index,kg/m2,"[1,4]",WHO BMI categories (1-4),Categorical BMI,Categorical body mass index,,,,active,,
```
`recStart` should be reviewed for this `HWTGBMI_cat4` variable

rafdoodle commented on inst/extdata/variable_details.csv:
```diff
@@ -1609,56 +1586,63 @@ HUPDPAD,HUPDPAD_cat5_5,cat,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p,
+HWTGBMI_cat4,HWTGBMI_cat4_NA::b,cat,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p",DerivedVar::[HWTGBMI],,cat,NA::b,4,missing,missing,kg/m2,NA::b,missing,Categorical BMI,Categorical body mass index,,,,active,,
+HWTGBMI_der,N/A,cont,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p","DerivedVar::[HWTGHTM, HWTGWTK]",,N/A,Func::calculate_bmi,N/A,N/A,N/A,kg/m2,"[15,50]",Derived BMI valid range,Derived BMI,Derived Body Mass Index,"BMI variable derived from the harmonized height and weig ...
+HWTGBMI_der,N/A,cont,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p","DerivedVar::[HWTGHTM, HWTGWTK]",,N/A,NA::b,N/A,missing,missing,kg/m2,N/A,N/A,Derived BMI,Derived Body Mass Index,,,,active,,No typeStart
+HWTGBMI_der_cat4,HWTGBMI_der_cat4N/A_Func::categorize_bmi,cat,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2011_2012_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2017_2018_p",DerivedVar::[HWTGBMI_der],,N/A,Func::categorize_bmi,N/A,N/A,N/A,kg/m2,N/A,N/A,Categorical BMI,Categorical body mass index,,,,active,,No typeStart
```
`categorize_` functions removed from `R/bmi.R` -> `cat4` rows in `variable_details.csv` need to be refactored

rafdoodle commented on inst/extdata/variables.csv:
```diff
@@ -1,380 +1,382 @@
+HUIDHSI_cat50,Overall HUI,Categorical Health Utility Index (HUI3),Categorical,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2009_m, cchs2010_m, cchs2012_m","cchs2001_p::HUIADHSI, cchs2003_p::HUICDHSI, cchs2005_p::HUIEDHSI, cchs2015_2016_p::HUIDVHSI, [HUIDHSI]",Health utility index,Health status,N/A, ...
+HUIGDEX,HUI Dexterity,"Dexterity trouble - function code (D, G)",Categorical,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2009_m, cchs2010_m, cchs2012_m","cchs2001_p::HUIAGDEX, cchs2003_p::HUICGDEX, cchs2005_p::HUIEGDEX, cchs2015_2016_p::HUIDGDEX, cchs2009_m::HUIDDEX, cchs2010_m::HUIDDEX, cchs2012_ ...
+HUIGHER,HUI Hearing,"Hearing problems - function code (D, G)",Categorical,"cchs2001_p, cchs2003_p, cchs2005_p, cchs2007_2008_p, cchs2009_2010_p, cchs2010_p, cchs2012_p, cchs2013_2014_p, cchs2014_p, cchs2015_2016_p, cchs2009_m, cchs2010_m, cchs2012_m","cchs2001_p::HUIAGHER, cchs2003_p::HUICGHER, cchs2005_p::HUIEGHER, cchs2015_2016_p::HUIDGHER, cchs2009_m::HUIDHER, cchs2010_m::HUIDHER, cchs2012_m:: ...
+HUIGMOB,HUI Mobility,"Mobility trouble - func
```
Add a Master file version of `HWTGBMI_cat4`?

DougManuel committed:
Fix HWTDHTM worksheet typos and add Master BMI functions

Worksheet fixes (HWTDHTM rows only):
- Fix HTWDVHTM typo → HWTDVHTM (4 rows in variable_details, 1 in variables)
- Fix cchs_2003_m → cchs2003_m, cchs_2005_m → cchs2005_m underscore typos
- Fix missing cchs prefix: 2015_2016_m → cchs2015_2016_m

New R/bmi-master.R:
- Self-contained Master-file BMI functions (calculate_bmi_D, adjust_bmi_D,
  categorize_bmi_D) that don't depend on assign_tagged_na
- Referenced by variable_details.csv as Func:: entries for HWTDBMI_der,
  HWTDCOR_der, HWTDBMI_der_cat4

DougManuel commented:
### Code review

Reviewed 10 BMI variables (HWTGHTM, HWTGWTK, HWTGBMI, HWTGBMI_der, HWTGBMI_der_cat4, HWTGBMI_cat4, HWTGCOR_der, HWTDBMI_der, HWTDBMI_der_cat4, HWTDCOR_der) for PUMF and Master across 2001–2018.

#### Issues found and fixed (68ee15db)

1. **HTWDVHTM typo in HWTDHTM variableStart** (L3, worksheet naming)
   4 rows in `variable_details.csv` and 1 row in `variables.csv` had transposed `HTWDVHTM` instead of `HWTDVHTM`. Confirmed via MCP cchs-metadata: `HWTDVHTM` exists, `HTWDVHTM` does not. **Fixed.**

2. **Underscore typos in database names** (L3, databaseStart consistency)
   `cchs_2003_m`, `cchs_2005_m` (with extra underscore) in HWTDHTM variableStart. **Fixed** → `cchs2003_m`, `cchs2005_m`.

3. **Missing cchs prefix** (L3, databaseStart consistency)
   `2015_2016_m` missing the `cchs` prefix in HWTDHTM variableStart. **Fixed** → `cchs2015_2016_m`.

4. **Missing Master BMI DV functions** (L4/L6, derived variable specification)
   `variable_details.csv` references `Func::calculate_bmi_D`, `Func::adjust_bmi_D`, `Func::categorize_bmi_D` but no R file defined them. Created `R/bmi-master.R` with self-contained implementations using `dplyr::case_when` and `haven::tagged_na`. These do not depend on `assign_tagged_na` (which is undefined on this branch). **Fixed.**

#### Issues noted but not fixed

5. **HWTGBMI_cat4 semantic change** (L2, confidence 75)
   `variableStart` changed from `DerivedVar::[HWTGBMI_der]` (cchsflow-derived BMI) to `DerivedVar::[HWTGBMI]` (StatCan's pre-calculated grouped BMI). This changes the input source — intentional? If so, no action needed.

6. **PUMF BMI functions broken** (pre-existing, confidence 100)
   `bmi_fun` → `calculate_bmi` → `calculate_bmi_core` → `assign_tagged_na` — the last function is undefined on `5yr-age-categories`. All PUMF BMI derived variables (`HWTGBMI_der`, `HWTGCOR_der`, `HWTGBMI_der_cat4`) fail at runtime. Pre-existing on target branch; not introduced by this PR.

7. **PUMF L6 integration blocked** (pre-existing)
   `rec_with_table()` runs but produces no BMI output columns due to issue 6. PUMF height/weight pass-through variables (HWTGHTM, HWTGWTK) work correctly. Master DV functions (issue 4 fix) tested with synthetic data — all produce correct results.

#### L6 integration summary

PUMF pass-through variables (HWTGHTM, HWTGWTK) recoded successfully across all 9 cycles. PUMF derived variables blocked by pre-existing `assign_tagged_na` issue. Master DV functions validated with synthetic data (BMI calculation, WHO 4-category classification, Connor Gorber bias correction all correct).

DougManuel committed:
feat: add v3 missing data infrastructure (L1-L6)

Add the 6-file v3 infrastructure stack for missing data handling:
- file-sourcing.R (L1): robust file sourcing with here()
- worksheet-loaders.R (L2B): worksheet metadata loading
- worksheet-getters.R (L3): get_variable_details(), get_variables()
- missing-pattern-cache.R (L4): session-level missing pattern cache
- missing-data-functions.R (L5): any_missing(), get_priority_missing(), assign_missing()
- clean-variables.R (L6): clean_variables(vars=, output_format=)

These files provide the shared infrastructure for v3 derive functions
across all variable domains (BMI, smoking, etc.).

DougManuel committed:
Merge branch 'feature/v3.0.0-validation-infrastructure' into adj-bmi

DougManuel committed:
refactor: rename old clean_variables() to clean_variables_legacy()

Resolve namespace collision with new clean_variables() from
R/clean-variables.R (v3 infrastructure). The old API using
continuous_vars/categorical_vars params is renamed to
clean_variables_legacy(). Callers in smoking.R and alcohol.R
will be updated when v3-smoking merges.

DougManuel committed:
feat(bmi): rewrite calculate_bmi to silver tier with 3-step architecture

Also fix is_value_in_range() and apply_else_rule() in clean-variables.R
to handle the copy_mapping format (recStart/recStart_values/recEnd) from
map_recStart_to_recEnd(), which was mismatched with the expected min/max
and action keys.

DougManuel committed:
feat(bmi): add silver-tier adjust_bmi with sex-specific bias correction

DougManuel committed:
feat(bmi): add silver-tier categorize_bmi with WHO categories

DougManuel committed:
feat(bmi): rewrite Master BMI functions to silver tier

DougManuel committed:
feat(bmi): rename _D to _master, update aliases and worksheet references

- Add deprecated aliases: calculate_bmi_D, adjust_bmi_D, categorize_bmi_D
  forwarding to _master equivalents
- Update variable_details.csv Func:: references from _D to _master
- Add 6 alias delegation tests (76 total passing)

DougManuel committed:
feat(bmi): unify PUMF and Master BMI into source-agnostic functions

Replace separate PUMF (bmi.R) and Master (bmi-master.R) implementations
with unified source-agnostic functions using semantic parameter names
(height_m, weight_kg). Worksheet routing maps PUMF grouped variables
(HWTGHTM/HWTGWTK) and Master continuous variables (HWTDHTM/HWTDWTK)
to the same function parameters.

- Delete bmi-master.R; consolidate into bmi.R
- Rename parameters from CCHS variable names to semantic names
- Update deprecated aliases (_D suffix) to point to unified functions
- Add @param ... to all three BMI functions for R CMD check compliance
- Update worksheets for unified function routing
- Regenerate man/ pages

DougManuel commented:
## BMI unification: source-agnostic functions

Replaced the separate PUMF (`bmi.R`) and Master (`bmi-master.R`) implementations with unified source-agnostic functions.

### What changed

- **Deleted `bmi-master.R`** — all three BMI functions (`calculate_bmi`, `adjust_bmi`, `categorize_bmi`) now live in `bmi.R` and work with both PUMF and Master data
- **Semantic parameter names** — `height_m`/`weight_kg` instead of CCHS variable names (`HWTGHTM`/`HWTGWTK`). The worksheet routes the appropriate source variables to these parameters.
- **Deprecated aliases updated** — `_D` suffix aliases now point to the unified functions (not the deleted `_master` variants)
- **`@param ...` added** — fixes R CMD check warnings caused by deprecated aliases using `function(...)` with `@rdname`
- **Worksheets updated** — both PUMF and Master databases route to the same functions via `variables.csv`/`variable_details.csv`

### Design rationale

Same pattern as `calculate_pack_years()` — the function doesn't need to know whether it's receiving PUMF grouped midpoints or Master continuous measurements. The worksheet handles the source variable mapping:

| Parameter | PUMF source | Master source |
|-----------|-------------|---------------|
| `height_m` | HWTGHTM (grouped) | HWTDHTM (continuous) |
| `weight_kg` | HWTGWTK (grouped) | HWTDWTK (continuous) |

### Verification

- 58/58 unit tests pass
- R CMD check: 0 new errors/warnings/notes (all pre-existing)

DougManuel committed:
style: normalize CSV quoting in variable_details.csv

Run fix-worksheets.R to remove unnecessary quotes per project convention.

rafdoodle committed:
feat(bmi): corrected labels for Master BMI and weight variable, added NA::a rows for all derived HWT variables, and eliminated redundant/incorrect HWTGBMI_cat4 variable

rafdoodle commented:
Changes manually merged to v3 via commit [3ce9a9f](https://github.com/Big-Life-Lab/cchsflow/commit/3ce9a9f389c737bdc43559eae5e53a3b78d506ad). Will close this PR now.
